File: 201-300/266. Palindrome Permutation.java
class Solution {
    public boolean canPermutePalindrome(String s) {
        Set<Character> set = new HashSet<Character>();
        int len = s.length();
        for(int i = 0;  i < len; i++){
            if(set.contains(s.charAt(i))) set.remove(s.charAt(i));
            else
                set.add(s.charAt(i));
        }
        return set.size() == 0 || set.size() ==1;
    }
}

// Topic: Hash Table
// Method: Use set
/* Explanation: iterate over string.
                add character to set if set does not contain that character
                remove character if set already contains that character
                check if the size equals 0 or 1
                0: even number of any character in the string
                1: even number of any character except one or just one character with odd number
*/

File: 201-300/266. Palindrome Permutation.py
class Solution:
    def canPermutePalindrome(self, s: str) -> bool:
        chance = 1
        count = collections.Counter(s)
        print(count)
        for letter in set(s):
            if count[letter] % 2 == 1:
                chance -= 1
                print(chance)
            if chance < 0:
                return False
        return True

    """
    The idea is simple: count the number of each character.
    and iterate through the count to check number
    if number is even, it always can be used
    if number is odd, it one of its character must be in the center.
    if another character wiht odd number shows up.
    Palindrome cannot be formed.
    """
